code/_core/obj/hud/inventory/interaction.dm

```diff
@@ -43,29 +43,46 @@
 		if(!isturf(caller.loc))
 			return TRUE
 		var/mob/living/L = caller
-		caller.face_atom(object)
-		var/atom/movable/object_to_throw = top_object
+		caller.face_atom(object) //Changing dir
+		var/atom/movable/object_to_throw
+		if(grabbed_object) object_to_throw = grabbed_object //We want to throw something that we pull
+		else object_to_throw = top_object //Else we throw something in our hand
+
 		if(istype(object_to_throw))
-			var/obj/item/I = object_to_throw
-			if(I.additional_clothing_parent || I.no_drop)
-				caller.to_chat(span("warning","You can't throw this!"))
-				return TRUE
-			var/vel_x = object.x - caller.x
+			if(ismob(object_to_throw) && grab_level < 2) //To throw mob you need agressive grab
+				caller.to_chat(span("warning","You need a better grip to do that!"))
+				return TRUE //Isn't agressive grab on mob?
+
+			var/vel_x = object.x - caller.x //Caller pos is the same as object anyway
 			var/vel_y = object.y - caller.y
 			var/highest = max(abs(vel_x),abs(vel_y))
 
-			if(!highest)
-				I.drop_item(get_turf(caller)) //Drop if we can't throw.
+			if(highest == 0) //Our position is the same as target
```

Comment:
This should be below the item checks because I.drop_item() won't get called on line 70 ever.

Reply:
Done.